### PR TITLE
Added support for screen and window dimension query in RNS shell.

### PR DESCRIPTION
**Pull request by munezbn**
PlatformDisplay Class will provide screen/graphics dimension
Window Class will provide their current dimension.

### CODE REVIEW

File: rns_shell/platform/graphics/libwpe/WindowLibWPE.cpp [hunk 1]
```diff
@@ -147,6 +149,8 @@ Window* Window::createNativeWindow(void* platformData) {
         delete window;
         return nullptr;
     }
+    if(!mainWindow_)
```
Comment (shefayaparwin) [inline rns_shell/platform/graphics/libwpe/WindowLibWPE.cpp:152]: As a TODO Action: does it Needed here  to check the created window type for Main Window

Comment (munezbn) [reply rns_shell/platform/graphics/libwpe/WindowLibWPE.cpp:152]: Ideally first window will be Main Window,  So if  it is `NULL`, it means there is no Main Window yet.. And we set `mainWindow` to `NULL` when we close/destroy the Main Window which would only happen when app exits.

File: rns_shell/platform/graphics/x11/WindowX11.h [hunk 2]
```diff
@@ -39,11 +39,16 @@ class WindowX11 : public Window {
             , visualInfo_(nullptr)
 #endif
             , MSAASampleCount_(1) {}
-    ~WindowX11() override { this->closeWindow(); }
+    ~WindowX11() override { if(this == mainWindow_) mainWindow_ = nullptr; this->closeWindow(); }
```
Comment (pprabhu-ng) [inline rns_shell/platform/graphics/x11/WindowX11.h]: minor comment : 
indentation of this function definition.
same for libwpe window destructor

Comment (munezbn) [reply rns_shell/platform/graphics/x11/WindowX11.h]: Will update

**Review by pprabhu-ng — APPROVED** [review]

**Pull request merged**